src/pytest_allure_spec_coverage/plugin.py

```diff
@@ -9,4 +9,166 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+
 """Main plugin module"""
+
+from dataclasses import dataclass, field
+from itertools import chain
+from typing import ClassVar, Iterable, List, Mapping
+
+import pytest
+from _pytest.config import Config
+from _pytest.config.argparsing import Parser
+from _pytest.config.exceptions import UsageError
+from _pytest.mark.structures import Mark
+from _pytest.nodes import Item
+from allure_commons.model2 import Label, Status, TestResult
+from allure_commons.reporter import AllureReporter
+from allure_commons.utils import uuid4
+from allure_pytest.listener import AllureListener
+from pluggy.manager import PluginManager
+
+from pytest_allure_spec_coverage.models.collector import Collector
+from pytest_allure_spec_coverage.models.scenario import Scenario
+from pytest_allure_spec_coverage.spec_collectors.sphinx import SphinxCollector
+
+
+def safe_get_marker(item: Item, name: str) -> Mark:
+    """Safely get pytest.Item marker"""
+
+    stub_marker = Mark(name="stub", args=(), kwargs={})
+    return item.get_closest_marker(name, stub_marker)
+
+
+def scenario_links(item: Item) -> Iterable[str]:
+    """Get scenario links from pytest.Item"""
+
+    return safe_get_marker(item, ScenariosMatcher.MARKER_NAME).args
+
+
+def allure_labels(scenario: Scenario) -> List[Label]:
+    """Make labels for Allure from the given scenario"""
+
+    labels = ("parentSuite", "suite", "subSuite")
```

Comment:
Use custom labels for implemented scenarios and custom + default for non implemented ones
@skhomuti  add custom labels names

Reply:
It seems to me that a list of these labels should be configurable from the `pyproject.toml` config

Reply:
Still relevant

Reply:
Added ConfigProvider.